### PR TITLE
fix(mobile): don't permanently cache an empty provider catalog

### PR DESCRIPTION
**Pull request by daniil-shumko**
## Summary

`SessionRepository` is a lazy singleton, so caching a provider-list response that carries no models would leave the model picker **blank forever** for that project.

Some backends build their model catalog asynchronously — the Cursor/ACP plugin warms it from an existing session only after the agent connects — so an early `listProviders` fetch can succeed with an empty `items`/`models` list. This change only caches a response when at least one provider actually has models, letting the next open retry and pick up the catalog once it's ready.

## Changes

- `session_repository.dart`: only populate `_providerCache` when the response contains at least one provider with a non-empty `models` map.
- `session_repository_test.dart`: regression test covering empty-then-populated fetches — the empty result is re-fetched (not cached), and the populated result is served from cache.

## Testing

- `dart test module_core` — all 408 tests pass.

---
## Summary by cubic
Stops permanently caching empty or partial provider catalogs so the model picker doesn’t stay blank. `SessionRepository` now caches provider lists only when the list is non-empty and every provider has models, allowing retries until the catalog is ready.

- **Bug Fixes**
  - Cache provider list only when non-empty and all providers have non-empty `models` in `session_repository.dart`.
  - Add tests to ensure empty and partially populated results aren’t cached, and fully populated results are.

<sup>Written for commit 7f7889f16f492c7aec1f32e41575907e333514f8. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/sesori-ai/sesori_apps_monorepo/pull/331?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: client/module_core/lib/src/repositories/session_repository.dart [hunk 1]
```diff
@@ -104,7 +104,15 @@ class SessionRepository {
...
-    if (response is SuccessResponse<ProviderListResponse>) {
+    // Only cache a response that actually carries models. Some backends build
+    // their model catalog asynchronously (e.g. the Cursor/ACP plugin warms it
+    // from an existing session after the agent connects), so an early fetch can
+    // succeed with an empty list. Caching that empty result — permanently, since
+    // this repository is a lazy singleton — would leave the model picker blank
+    // forever for that project. Skipping the cache lets the next open retry and
+    // pick up the catalog once it is ready.
+    if (response is SuccessResponse<ProviderListResponse> &&
+        response.data.items.any((provider) => provider.models.isNotEmpty)) {
```
[the remark this answers is not in the document]
Comment (daniil-shumko) [reply client/module_core/lib/src/repositories/session_repository.dart]: [Sesori reply] Addressed (in commit 7f7889f1)

Switched the cache guard from `any` to `isNotEmpty && every`, so a response is only cached once every provider carries models. A partially-populated catalog (one provider still warming with an empty `models` map) is no longer cached, and the next open refetches until the full catalog is ready. Added a regression test covering the multi-provider partial case.

File: client/module_core/lib/src/repositories/session_repository.dart [hunk 2]
```diff
@@ -104,7 +104,15 @@ class SessionRepository {
...
       _providerCache[projectId] = response.data;
     }
```
[the remark this answers is not in the document]
Comment (daniil-shumko) [reply client/module_core/lib/src/repositories/session_repository.dart:122]: [Sesori reply] Addressed (in commit 7f7889f1)

Applied your suggestion: the guard is now `response.data.items.isNotEmpty && response.data.items.every((provider) => provider.models.isNotEmpty)`, which prevents the partial-caching bug in the multi-provider (OpenAI populated + Cursor warming) scenario while still rejecting a fully empty list via the vacuous-truth `isNotEmpty` check. Added a regression test for the partial-population case.
